iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -259,7 +265,7 @@ class optional : public FunctionalInterface<optional<T>, T, void>
 ///             of T
 /// @return optional which contains T constructed with args
 template <typename OptionalBaseType, typename... Targs>
-optional<OptionalBaseType> make_optional(Targs&&... args) noexcept;
+auto make_optional(Targs&&... args) noexcept -> optional<OptionalBaseType>;
```

Comment:
Please revert this. Here we have to write more and destroy the normal reading flow to follow a rule.

The rational behind the rule is that this prevents `typename` usage. But here is no `typename` and it also does not prevent the `typename` usage. The only thing which happens here is that the code becomes more complex and the normal reading flow of `returntype`, `functionname()`  and `qualifiers` is destroyed.